spec/models/a_document_subclass.rb

```diff
@@ -0,0 +1,33 @@
+RSpec.shared_examples "a Document subclass" do
+  describe "#save!" do
+    it "saves a valid document" do
+      publishing_api_has_item(payload)
+      Timecop.freeze(Time.parse("2015-12-18 10:12:26 UTC"))
+      stub_any_publishing_api_put_content
+      stub_any_publishing_api_patch_links
+
+      instance = described_class.find(payload["content_id"])
+      expect(instance.save!).to eq(true)
+
+      expected_payload_sent_to_publishing_api = saved_for_the_first_time(write_payload(payload))
+
+      assert_publishing_api_put_content(instance.content_id, expected_payload_sent_to_publishing_api)
+      expect(expected_payload_sent_to_publishing_api.to_json).to be_valid_against_schema('specialist_document')
+    end
+
+    def saved_for_the_first_time(document)
+      timestamp = Time.now.to_datetime.rfc3339
+      document.deep_merge(
+        "public_updated_at" => timestamp,
+        "details" => {
+          "change_history" => [
+            {
+              "public_timestamp" => timestamp,
+              "note" => "First published.",
+            }
+          ]
+        }
+      )
+    end
```

Comment:
Should we use this method in the `document_spec` too? There's a similar assertion about `First published`.